plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -1021,61 +1021,67 @@ public function get_coupon_message( $msg_code ) {
 	public function get_coupon_error( $err_code ) {
 		switch ( $err_code ) {
 			case self::E_WC_COUPON_INVALID_FILTERED:
-				$err = __( 'Coupon is not valid.', 'woocommerce' );
+				/* translators: %s: coupon code */
+				$err = sprintf( esc_html__( 'Coupon %s is not valid.', 'woocommerce' ), '"' . esc_html( $this->get_code() ) . '"' );
```

Comment:
```suggestion
				$err = sprintf(
					esc_html__( 'Coupon "%s" is not valid.', 'woocommerce' ),
					esc_html( $this->get_code() )
				);
```

One thing I would recommend is making the `sprintf` multi-line. It's way easier to read that way. Also, I think the quotes would be better in the translatable string rather than in the token. Some languages use other characters besides quotes for things like this.

Reply:
> Some languages use other characters besides quotes for things like this.

That's a good point, I didn't realize that. 

I actually had them in the string originally, but moved them out since `esc_html__` was replacing them with `&quot;`. 

This seems fine for some of the respones, eg.

Before changes
```
{
"error": "woocommerce_rest_invalid_coupon",
"message": "The minimum spend for this coupon is <span class=\"woocommerce-Price-amount amount\"><bdi><span class=\"woocommerce-Price-currencySymbol\">&#36;</span>50.00</bdi></span>."
}
```

After changes
```
{
"error": "woocommerce_rest_invalid_coupon",
"message": "The minimum spend for coupon &quot;20_dollars_off_cart_min_amount_50&quot; is <span class=\"woocommerce-Price-amount amount\"><bdi><span class=\"woocommerce-Price-currencySymbol\">&#036;</span>50.00</bdi></span>."
}
```

Since the message already contains html entities it seems ok to assume the client will decode them. (I personally believe this doesn't belong to REST API, but that's another story)

However, for other messages, this assumption seems quite risky, for example:

Before changes
```
{
"error": "woocommerce_rest_invalid_coupon",
"message": "Coupon \"wrong_coupon_code\" does not exist!"
}
```

After changes
```
{
"error": "woocommerce_rest_invalid_coupon",
"message": "Coupon &quot;wrong_coupon_code&quot; does not exist!"
}
```


Wdyt @coreymckrill ?

Reply:
🤔 What makes the second example risky? Is it that the `&quot;` HTML entity might be interpreted as part of the coupon code?

Reply:
Just the fact that the message contains a HTML entity now even thought it didn't contain it before => the client needs to interpret it as HTML instead of just a plain text.

Reply:
Ah, I see what you mean. I think it's a fairly common pattern in our APIs to have HTML entities like this because of escaping, so it may be worth adapting the client to be able to handle it. But a couple of other options:

* Use `wp_kses` with an empty array for the "allowed tags" parameter, so that all HTML tags get stripped (but quotes don't get converted)
* Don't wrap the coupon code in quotes at all

Reply:
👍 Let's keep it simple and remove the quotes. They don't seem necessary anyway and will reduce risks of introducing some issues in translations.

Reply:
Actually, removing the quotes might not be the best option - we are using them across the codebase, even for successful messages. Considering that in my tests the quotes didn't cause any troubles on the UI, let's continue escaping them as we have been escaping them. The worst case scenario, they won't be de-escaped which is not a great UX but also not a critical error.